From 1f34530921070bb5c150e44099c7948b9af61fd2 Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Mon, 29 Sep 2014 10:22:52 +0800
Subject: [PATCH 0134/5242] MLK-9617-11 usb: common: otg: use a_set_b_hnp_en
 instead of b_hnp_enable

commit  d8469a38010801839c88b5282e7b026e8b9c88ed from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch replaces b_hnp_enable with a_set_b_hnp_en to comply with state diagram
of USB and EH spec.

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit 4734ed1f813959a7544ad744d60ab119efb04ba9)
(cherry picked from commit 853e4e0a2c59f034b177eebd3e7254efc8feffd4)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/common/usb-otg-fsm.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index 84f46d3..00a6966 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -440,9 +440,9 @@ int otg_statemachine(struct otg_fsm *fsm)
 			otg_set_state(fsm, OTG_STATE_A_VBUS_ERR);
 		break;
 	case OTG_STATE_A_SUSPEND:
-		if (!fsm->b_conn && fsm->otg->host->b_hnp_enable)
+		if (!fsm->b_conn && fsm->a_set_b_hnp_en)
 			otg_set_state(fsm, OTG_STATE_A_PERIPHERAL);
-		else if (!fsm->b_conn && !fsm->otg->host->b_hnp_enable)
+		else if (!fsm->b_conn && !fsm->a_set_b_hnp_en)
 			otg_set_state(fsm, OTG_STATE_A_WAIT_BCON);
 		else if (fsm->a_bus_req || fsm->b_bus_resume)
 			otg_set_state(fsm, OTG_STATE_A_HOST);
-- 
1.7.9.5

